docs/docs/client-only-routes-and-user-authentication.md

```diff
@@ -34,20 +128,34 @@ exports.onCreatePage = async ({ page, actions }) => {
 > 💡 Note: There's also a plugin to simplify the creation of client-only routes in your site:
 > [gatsby-plugin-create-client-paths](/packages/gatsby-plugin-create-client-paths/).
 
+The above code (as well as the `gatsby-plugin-create-client-paths` plugin) updates the `/app` page at build time to add the `matchPath` parameter in the page object to make it so that the configured pages (in this case, everything after `/app`, like `/app/dashboard` or `/app/user`) can be navigated to by Reach Router.
+
+_Without_ this configuration set up, a user that clicks on a link to `<yoursite.com>/app/user` will instead be routed to the static `/app` page instead of the component or page you have set up at `/app/user`.
+
 > Tip: For applications with complex routing, you may want to override Gatsby's default scroll behavior with the [shouldUpdateScroll](/docs/browser-apis/#shouldUpdateScroll) Browser API.
 
-Check out the ["simple auth" example site](https://github.com/gatsbyjs/gatsby/blob/master/examples/simple-auth/) for a demo implementing user authentication and restricted client-only routes.
+## Configuring and handling client-only routes on a server
 
-## Configuring a Server to Handle Client Side Routes
+If you are hosting on your own server, you can opt to configure the server to handle client-only routes instead of using the `matchPath` method already explained.
```

Comment:
```suggestion
If you are hosting on your own server, you can opt to configure the server to handle client-only routes instead of using the `matchPath` method explained above.
```